src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -144,9 +144,36 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
     }
   }
 
+  def updateAccess(docs: LibrarySearchResponse, workspaces: Seq[WorkspaceListResponse]) = {
+    val ids = workspaces collect {
+      case workspaceResponse: WorkspaceListResponse if (workspaceResponse.accessLevel != WorkspaceAccessLevels.NoAccess) =>
+        workspaceResponse.workspace.workspaceId
+    }
+
+    val updatedResults = docs.results.map { document =>
+      val docId = document.asJsObject.fields.get("workspaceId")
+      val newJson = docId match {
+        case Some(id) =>  val workspaceId = id.toString.replace("\"", ""); document.asJsObject.fields + ("workspaceAccess" -> JsBoolean(ids.contains(workspaceId)))
+        case None => document.asJsObject.fields + ("workspaceAccess" -> JsBoolean(true)) // is this the right thing to do?
```

Comment:
in the UI, I think the right thing to do is to send the user to the workspace and let Rawls decide what to do.

In the API, I think the right thing to do is to have three states: `true` if the user has access, `false` if the user does not, and empty/null/missing if we can't determine access. So how about:
```
case _ => document
```
which would leave the `workspaceAccess` field out of the response entirely? This would of course require corresponding logic in the UI to handle the missing value.